chia/full_node/mempool_check_conditions.py

```diff
@@ -132,40 +134,222 @@ def mempool_assert_my_amount(condition: ConditionWithArgs, unspent: CoinRecord)
+    seconds = args.first().atom
+    seconds_int = int_from_bytes(seconds)
+    # this condition is inherently satisified, there is no need to keep it
+    if seconds_int <= 0:
+        return None
+    if seconds_int >= 2 ** 64:
+        raise ValidationError(error_code)
+    # note that this may change the representation of seconds. If the original
+    # buffer had redundant leading zeroes, they will be stripped
+    return [int_to_bytes(seconds_int)]
+
+
+def parse_height(args: SExp, error_code: Err) -> Optional[List[bytes]]:
+    height = args.first().atom
+    height_int = int_from_bytes(height)
+    # this condition is inherently satisified, there is no need to keep it
+    if height_int <= 0:
+        return None
+    if height_int >= 2 ** 32:
+        raise ValidationError(error_code)
+    # note that this may change the representation of the height. If the original
+    # buffer had redundant leading zeroes, they will be stripped
+    return [int_to_bytes(height_int)]
+
+
+def parse_fee(args: SExp) -> List[bytes]:
+    fee = args.first().atom
+    fee_int = int_from_bytes(fee)
+    if fee_int >= 2 ** 64 or fee_int < 0:
+        raise ValidationError(Err.RESERVE_FEE_CONDITION_FAILED)
+    # note that this may change the representation of the fee. If the original
+    # buffer had redundant leading zeroes, they will be stripped
+    return [int_to_bytes(fee_int)]
+
+
+def parse_coin_id(args: SExp, error_code: Err) -> List[bytes]:
+    coin = args.first().atom
+    if len(coin) != 32:
+        raise ValidationError(error_code)
+    return [coin]
+
+
+def parse_hash(args: SExp, error_code: Err) -> List[bytes]:
+    h = args.first().atom
+    if len(h) != 32:
+        raise ValidationError(error_code)
+    return [h]
+
+
+def parse_amount(args: SExp) -> List[bytes]:
+    amount = args.first().atom
+    amount_int = int_from_bytes(amount)
+    if amount_int < 0:
+        raise ValidationError(Err.ASSERT_MY_AMOUNT_FAILED)
+    if amount_int >= 2 ** 64:
+        raise ValidationError(Err.ASSERT_MY_AMOUNT_FAILED)
+    # note that this may change the representation of amount. If the original
+    # buffer had redundant leading zeroes, they will be stripped
+    return [int_to_bytes(amount_int)]
+
+
+def parse_announcement(args: SExp) -> List[bytes]:
+    msg = args.first().atom
+    if len(msg) > 1024:
+        raise ValidationError(Err.INVALID_CONDITION)
+    return [msg]
+
+
+def parse_condition_args(args: SExp, condition: ConditionOpcode) -> Tuple[int, Optional[List[bytes]]]:
+    """
+    Parse a list with exactly the expected args, given opcode,
+    from an SExp into a list of bytes. If there are fewer or more elements in
+    the list, raise a RuntimeError. If the condition is inherently true (such as
+    a time- or height lock with a negative time or height, the returned list is None
+    """
+    if condition is ConditionOpcode.AGG_SIG_UNSAFE or condition is ConditionOpcode.AGG_SIG_ME:
```

Comment:
You might consider using a dictionary with type `Dict[ConditionOpcode, Tuple[int, Callable[[SExp, Err], Tuple[int, Optional[List[bytes]]]]` populated with things like

```
{
   ConditionOpcode.CREATE_COIN: (ConditionCost.CREATE_COIN.value, parse_create_coin),
 ...
}
```

It might make the code a bit easier to read & expand.

Reply:
I just experimented with this approach. I don't think it's a clear improvement. I'd have to use `functools.partial()` to lock in the error code argument to some of these functions. The main noise comes from the repetition of the names of the constants, e.g. `ConditionOpcode.ASSERT_SECONDS_ABSOLUTE`, `ConditionCost.ASSERT_SECONDS_ABSOLUTE.value` and `Err.ASSERT_SECONDS_ABSOLUTE_FAILED`.

I think the main risk with this code is that some of these constants don't match correctly. I made a commit where I abbreviated the enum names.

I think making a dict with function pointers would be slightly less straight forward than the existing code.